packages/next/build/webpack/plugins/flight-types-plugin.ts

```diff
@@ -25,8 +25,9 @@ type TEntry = typeof entry
 
 check<IEntry, TEntry>(entry)
 
-type PageProps = { params?: any }
-type LayoutProps = { children: React.ReactNode; params?: any }
+type PageParams = Record<string, string>
+type PageProps = { params?: PageParams }
```

Comment:
There can also be `searchParams` x-ref: https://github.com/vercel/next.js/blob/0a97d07802d1fc5c98b32a74bcc7d94dd3e61b36/packages/next/server/app-render.tsx#L1191